packages/@aws-cdk/aws-lambda/lib/lambda.ts

```diff
@@ -245,6 +266,7 @@ export class Function extends FunctionRef {
       functionName: props.functionName,
       description: props.description,
       code: new cdk.Token(() => props.code.toJSON()),
+      layers: props.layerVersions && props.layerVersions.map(layer => layer.layerVersionArn),
       handler: props.handler,
       timeout: props.timeout,
       runtime: props.runtime.name,
```

Comment:
Would be very useful to support “lambda.addLayer” which probably implies the synthesis of “layers” should be lazy

Reply:
If you can `addLayer` after create, then you can also `addCode` after create. One does not make sense without the other.

Reply:
I disagree. Code is a required property. You can't have a Lambda function without code. Layers are optional, and I can think of cases where someone would want to mutate a Lambda function after it was created to add layers to it (but code - less likely).

Reply:
What's the point of adding layers after the code? The code depends on the layers, or the layers are useless.

Reply:
Here's an example: let's say I want to vend a construct and I want to also vend a runtime library with it, so that people who consume this construct will be able to use this library in their lambda functions. 

Let's say I want to implement an API like so:

```ts
myConstruct.install(lambda);
```

The implementation will be something like:

```ts
install(fn: lambda.Function) {
  fn.addToRolePolicy(...); // add permissions for this lambda function to access the resources
  fn.addLayer(...); // add a layer to the function so that my runtime library can be used
  fn.addEnvironment(...); // add some environment
}
```

I think this is a very powerful pattern, and I hope we will be able to formalize it even more (in the form of something like: `lambda.install(module)` (inverted control). But the basic idea is that the Lambda's environment should be mutable in the CDK.

Reply:
Sounds like future-proofing to me (and that's a feature that can easily be added later).

Reply:
I don't think it's future proofing. It's about designing the API for Lambda layers and trying to understand the use cases in which it will be used. I am arguing that it would be a very common situation to use layers in this "inverted control" scenario. That's essentially what layers are about.